### ISSUE TITLE
Conflict with Wooden Hoppers 

### ISSUE THREAD

**Issue by Snarphe** [AbdElAziz333/Canary#46]
### Version

> canary-mc1.19.2-0.0.10.jar 
### Reproduction Steps

> 1. Place a Wooden Hopper
> 2. Place a Redstone Comparator next to and facing away from the Wooden Hopper
> 3. Minecraft crashes

![image](https://user-images.githubusercontent.com/69836540/201222378-162976de-d206-4059-b6d1-ab914ec7168a.png)

### Additional Information

> woodenhopper-1.19-1.3.2.0.jar
> https://github.com/cech12/WoodenHopper

[crash-2022-11-10_14.26.58-server.txt](https://github.com/AbdElAziz333/Canary/files/9985125/crash-2022-11-10_14.26.58-server.txt)

**Comment by AbdElAziz333** [comment]
fixed in the coming update.

**Comment by AbdElAziz333** [comment]
can you see if the issue is fixed in the [build](https://www.curseforge.com/minecraft/mc-mods/canary/files/4091544)?

**Comment by Snarphe** [comment]
Tried that version, which crashed on launch, and the new one which did launch but still has the crash with wooden hoppers unfortunately

**Comment by cech12** [comment]
The issue was, that my mod only used the content of the new CanaryList (and not the list itself) to update the item list of the hopper. After that it called `this.setChanged` with that your mod tries to set a new CanaryList to the hopper. So an endless loop was the result.

I fixed that on my side by not calling `this.setChanged` any longer. Fixed versions are: [1.19-1.3.2.1](https://github.com/cech12/WoodenHopper/releases/tag/1.19-1.3.2.1) and [1.18.1-1.2.2.1](https://github.com/cech12/WoodenHopper/releases/tag/1.18.1-1.2.2.1)
